### PR TITLE
Unit test for method `Storage.Connection`

### PR DESCRIPTION
**Pull request by tisnik**
# Description

Unit test for method `Storage.Connection`

## Type of change

- Unit tests (no changes in the code)

## Testing steps

Done on CI

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by Bee-lee — APPROVED** [review]

**Review by dpensi — APPROVED** [review]

File: storage_test.go [hunk 1]
```diff
@@ -862,3 +862,24 @@ func TestClose(t *testing.T) {
...
+
+	// try to retrieve connection
+	returned := storage.Connection()
+	assert.NotNil(t, returned)
+
+	// try to close the storage
+	err = storage.Close()
+
+	// it should not fail
+	assert.Nil(t, err)
```
Comment (juandspy) [inline storage_test.go:884]: `assert.NoError` would be better

File: storage_test.go [hunk 2]
```diff
@@ -862,3 +862,24 @@ func TestClose(t *testing.T) {
...
+		Driver:        "sqlite3",
+		LogSQLQueries: false,
+	})
+
+	// we just happen to make connection without trying to actually connect
+	assert.Nil(t, err)
+
+	// try to retrieve connection
+	returned := storage.Connection()
+	assert.NotNil(t, returned)
```
Comment (juandspy) [inline storage_test.go:878]: `assert.Error` would be better

File: storage_test.go [hunk 3]
```diff
@@ -862,3 +862,24 @@ func TestClose(t *testing.T) {
...
+
+// TestConnection function checks the method Storage.Connection.
+func TestConnection(t *testing.T) {
+	storage, err := main.NewStorage(main.StorageConfiguration{
+		Driver:        "sqlite3",
+		LogSQLQueries: false,
+	})
+
+	// we just happen to make connection without trying to actually connect
+	assert.Nil(t, err)
```
Comment (juandspy) [inline storage_test.go:874]: `assert.NoError` would be better

File: storage_test.go (hunk 2 above)
Comment (tisnik) [reply storage_test.go:878]: it is not value of type `error` :)

File: storage_test.go (hunk 1 above)
Comment (tisnik) [reply storage_test.go]: fixed

File: storage_test.go (hunk 3 above)
Comment (tisnik) [reply storage_test.go]: fixed

**Review by epapbak — APPROVED** [review]
ok

**Pull request merged**